### PR TITLE
feat(EMS-2042): Name on policy, Different name on policy - Save and back

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction ✏️ 
This PR adds save and go back functionality for different name on policy and name on policy

## Resolution ✔️ 
* Added save and go back functionality
* Added unit and e2e tests
* Added mapSubmittedData condition for if empty string

### CODE REVIEW

File: src/ui/server/controllers/insurance/policy-and-export/different-name-on-policy/save-and-back/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  const mockFormBody = {
+    _csrf: '1234',
+    [FIRST_NAME]: mockContact[FIRST_NAME],
+  };
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    res.locals.application = mockApplication;
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/policy-and-export/different-name-on-policy/save-and-back/index.test.ts:37]: No need for this anymore as it's in `mockRes`

File: src/ui/server/controllers/insurance/policy-and-export/name-on-policy/save-and-back/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  const mockFormBody = {
+    _csrf: '1234',
+    [NAME]: SAME_NAME,
+  };
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    res.locals.application = mockApplication;
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/policy-and-export/name-on-policy/save-and-back/index.test.ts:39]: Should work without this line

File: src/ui/server/controllers/insurance/policy-and-export/name-on-policy/save-and-back/index.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  const { referenceNumber } = req.params;
+
+  const payload = constructPayload(req.body, FIELD_IDS);
+
+  const validationErrors = generateValidationErrors(payload);
+
+  try {
+    // // save the application
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/policy-and-export/name-on-policy/save-and-back/index.ts:33]: `// //`

File: e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/save-and-back.spec.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    });
+
+    it(`should redirect to ${ALL_SECTIONS}`, () => {
+      cy.assertUrl(allSectionsUrl);
+    });
+
+    it('should have the originally submitted answer selected when going back to the page after submission', () => {
+      cy.navigateToUrl(url);
+
+      // submit the form via 'save and go back' button
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/save-and-back.spec.js:88]: Not a blocker, can be fixed in next branch - typo? no submission here 😛

File: e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/save-and-back.spec.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/save-and-back.spec.js:108]: Same as above

**Review by ttbarnes — APPROVED** [review]

File: e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/save-and-back.spec.js (hunk 4 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/policy-and-exports/name-on-policy/save-and-back.spec.js:88]: next PR

**Pull request merged**